### DISCUSSION TITLE
How can I send headers for the handshake with the browser APIs WebSocket?

### DISCUSSION THREAD

**Discussion by jonasfroeller** [socketio/socket.io#5002] [Q&A] [answered]
I would like to do, what socket.io does, but I want to do it without socket.io, because I do not need all the other stuff, and I do not want to increase traffic by using it just for that.
Does anybody know how to do that? I searched for a solution, but I can not find out, how to do it. I know that Postman can do it, meaning it has to be possible and in the official definition it sounds like it is.

**Comment by darrachequesne** [comment, marked as answer]
Hi!

> How can I send headers for the handshake with the browser APIs WebSocket?

This is not possible, as the WebSocket API in the browser does not provide a way to send HTTP headers. Postman can do it because it's a Node.js client under the hood.

References: 

- https://developer.mozilla.org/en-US/docs/Web/API/WebSocket
- https://socket.io/docs/v4/client-options/#extraheaders

**Reply by jonasfroeller** [reply]
Thank you for the quick response! 
Might switch to socket.io now, 
although I do not need all the features.
But you have made a lot of sdks 
which would probably help me a lot,
because I am building the server 
for php, typescript and swift.
And you have implemented authentication,
which can only be implemented in cursed ways 
using the browser API.
Thank you for the library!

**Comment by guest271314** [comment]
Like this https://github.com/guest271314/direct-sockets-http-ws-server

**Reply by guest271314** [reply]
@darrachequesne This is possible using Direct Socket `TCPServerSocket`.
